upf/io/pddl_writer.py

```diff
@@ -34,6 +34,34 @@ def convert(self, expression):
         """Converts the given expression to a PDDL string."""
         return self.walk(self.simplifier.simplify(expression))
 
+    def walk_exists(self, expression, args):
+        assert len(args) == 1
+        vars = expression.variables()
+        vars_string = "("
+        for v in vars:
+            vars_string = vars_string + "?"
+            vars_string = vars_string + v.name()
+            vars_string = vars_string + " - "
+            vars_string = vars_string + str(v.type())
+        vars_string = vars_string + ")"
```

Comment:
Use f-strings instead of string concatenation

Reply:
```

    def walk_exists(self, expression, args):
        assert len(args) == 1
        vars = expression.variables()
        vars_string_list = []
        for v in vars:
            vars_string = f"?{v.name()} - {str(v.type())}"
            vars_string_list.append(vars_string)
        return f'(exists ({" ".join(vars_string_list)})\n {args[0]})'

```

Let me know if this is ok. Looks more elegant to me.